benches/classic_crypto.rs

```diff
@@ -0,0 +1,133 @@
+    const RSA_N: &str = "432529019456174073628056731021899753880199292843627050477235451320968504136109996834942250928981978445989472551473247465106240932979604095669286519963140687101286153803154189345553920544182137022020540002491541180583190915417665399496578760830730904289091934652022809043462927391234535724121396575445021309223";
+    const RSA_PK: &str = "272636431233265956354044639799116206612921445708076864293675274944980833816969100954445036939076331379279263791466802764599215907676388073791876743514004914016502795361305034353643235925659883900363706630095745283005183368525178846360740971098036527594113502421475552957731655910068081721705039500289221854513";
+
+    /// Naively generates an RSA key pair `(N, pk, sk)` with the provided `security_lvl`,
+    /// where `N` is the [`Modulus`] and `pk` and `sk` integer s.t. `(m^(pk * sk)) = m mod N`.
+    ///
+    /// This algorithm is naive, because it samples primes in a naive way,
+    /// i.e. rejection sampling uniformly at random and checking whether that sample is prime.
+    ///
+    /// The following steps are made:
+    /// 1. Generate two unequal primes `p` and `q` roughly of bit size `security_lvl / 2`
+    /// (s.t. `N` is roughly of bit size `security_lvl`).
+    /// 2. Compute `N = p * q`.
+    /// 3. Compute Euler's Phi function `phi(N) = (p-1) * (q-1)`.
+    /// 4. Choose `pk = 65537 = 0x10000000000000001` statically as public key (common as `enc` is very
+    /// efficient with this key, has sufficient size, and guarantees to have an inverse).
+    /// 5. Calculate `sk = pk^(-1) mod phi(N)`.
+    /// 6. Output `(N, pk, sk)`.
+    pub fn gen(security_lvl: u32) -> (Modulus, Z, Z) {
+        let lower_bound = Z::from(2).pow(security_lvl / 2).unwrap();
+        let upper_bound = Z::from(2).pow(security_lvl / 2 + 1).unwrap();
+
+        let p = sample_prime_naive(&lower_bound, &upper_bound);
+        let q = sample_prime_naive(&lower_bound, &upper_bound);
+        assert_ne!(p, q);
+
+        let modulus = Modulus::try_from(&(&p * &q)).unwrap();
+        let phi_mod = (&p - Z::ONE) * (&q - Z::ONE);
+
+        // standard prime value chosen as public key
+        // to remove necessity of choosing it at random
+        let pk = Z::from(65537);
+        let sk = Z::from(
+            Zq::try_from_z_z(&pk, &phi_mod)
+                .unwrap()
+                .inverse()
+                .expect("There must an inverse of this element as pk is prime."),
+        );
+
+        (modulus, pk, sk)
+    }
+
+    /// Returns a pre-computed RSA key pair that was computed with `gen(1024)`.
+    pub fn static_gen() -> (Modulus, Z, Z) {
+        let modulus = Modulus::from_str(RSA_N).unwrap();
+        let pk = Z::from(65537);
+        let sk = Z::from_str(RSA_PK).unwrap();
+        (modulus, pk, sk)
+    }
+
+    /// Encrypts a `msg` assumed to be smaller than `modulus` via RSA's `pk`.
+    ///
+    /// `RSA.Enc(N, pk, msg) = msg^pk mod N`
```

Comment:
`RSA.Enc` looks weird in rust.

```suggestion
    /// `enc(N, pk, msg) = msg^pk mod N`
```
or 
```suggestion
    /// `rsa_textbook::enc(N, pk, msg) = msg^pk mod N`
```